### PR TITLE
fix injectDatabaseUrl errors

### PR DESCRIPTION
**Pull request by jni2000**
fix injectDatabaseUrl errors found CommScope.

### CODE REVIEW

File: src/harness/testcases/WINNF_FT_S_WDB_testcase.py [hunk 1]
```diff
@@ -305,7 +305,7 @@ def test_WINNF_FT_S_WDB_2(self, config_filename):
     cpi_database.setFilesToServe(files)
 
     # Inject the CPI database URL into the SAS UUT.
-    self._sas_admin.InjectDatabaseUrl(config['cpiDatabaseConfig']['indexUrl'])
+    self._sas_admin.InjectDatabaseUrl({'type': 'CPI', 'url': config['cpiDatabaseConfig']['indexUrl']})
```
Comment (kate-harrison) [inline src/harness/testcases/WINNF_FT_S_WDB_testcase.py]: I think we need one more fix here so that we are sure to inject the full URL, not just the relative path, and then you will have addressed all of Javi's bugs :)

File: src/harness/testcases/WINNF_FT_S_WDB_testcase.py [hunk 2]
```diff
@@ -160,8 +160,8 @@ def test_WINNF_FT_S_WDB_1(self, config_filename):
                                 config['palDatabaseConfig']['filePath'])
 
     # Inject the PAL database URL into the SAS UUT.
-    self._sas_admin.InjectDatabaseUrl(pal_database.getBaseUrl()+
-                                      config['palDatabaseConfig']['fileUrl'])
+    self._sas_admin.InjectDatabaseUrl({'type': 'PAL', 'url': pal_database.getBaseUrl()+
+                                       config['palDatabaseConfig']['fileUrl']})
```
Comment (kate-harrison) [inline src/harness/testcases/WINNF_FT_S_WDB_testcase.py]: Looks like we need to append the file name to this URL as well (thanks Javi!).

Comment (jni2000) [reply src/harness/testcases/WINNF_FT_S_WDB_testcase.py]: The fileUrl already included the file name in it.

File: src/harness/testcases/WINNF_FT_S_WDB_testcase.py (hunk 1 above)
Comment (jni2000) [reply src/harness/testcases/WINNF_FT_S_WDB_testcase.py]: fixed.

**Comment by jni2000** [comment]
Kate and Javi

Fixed the CPI database full path issue. The PAL database url path already included the file name. Please can you do a double check before merge ?

Thanks

**Comment by kate-harrison** [comment]
James, I think you just need to change the 'filePath' in the PAL DB config to point to a file rather than a folder. So something like "/pal_db.json" would work.

File: src/harness/testcases/WINNF_FT_S_WDB_testcase.py [hunk 3]
```diff
@@ -161,7 +161,7 @@ def test_WINNF_FT_S_WDB_1(self, config_filename):
 
     # Inject the PAL database URL into the SAS UUT.
     self._sas_admin.InjectDatabaseUrl({'type': 'PAL', 'url': pal_database.getBaseUrl()+
-                                       config['palDatabaseConfig']['fileUrl']})
+                                       config['palDatabaseConfig']['filePath']})
```
Comment (kate-harrison) [inline src/harness/testcases/WINNF_FT_S_WDB_testcase.py]: I think you need to change the value of 'fileUrl', not this line. I understand 'filePath' to be the local path to the file, not the URL that the DB server will be expecting. So please revert this change and update 'fileUrl' to be 'pal_db.json'.

Comment (jni2000) [reply src/harness/testcases/WINNF_FT_S_WDB_testcase.py]: yes, you are right.

**Comment by jni2000** [comment]
Please take a look of both WDB.1 and .2 injected URL now, they shall be like

{ "type": "PAL", "url": "https://10.250.12.169:8003/rest/pal/v1/pal_db.json"}
{ "type": "CPI", "url": "https://10.250.12.169:8003/rest/cpi/v1/index.csv"}

File: src/harness/testcases/WINNF_FT_S_WDB_testcase.py (hunk 2 above)
Comment (jni2000) [reply src/harness/testcases/WINNF_FT_S_WDB_testcase.py]: I was wrong, now fixed.

File: src/harness/testcases/WINNF_FT_S_WDB_testcase.py [hunk 4]
```diff
@@ -231,8 +231,7 @@ def generate_WDB_2_default_config(self, filename):
             'status': 'ACTIVE',
             'publicKeyIdentifierFile': cpi_device_d_public_key_relative_file_path
         }],
-        'indexUrl':
-            os.path.join('testcases', 'testdata', 'cpi_db', 'index.csv')
+        'indexUrl': '/rest/cpi/v1/index.csv')
```
[the remark this answers is not in the document]
Comment (jni2000) [reply src/harness/testcases/WINNF_FT_S_WDB_testcase.py]: great catch. Please take a look of the fix.

File: src/harness/testcases/WINNF_FT_S_WDB_testcase.py [hunk 5]
```diff
@@ -285,8 +286,8 @@ def test_WINNF_FT_S_WDB_2(self, config_filename):
 
     # Create the main CPI database file.
     cpis_keys = ['cpiId', 'status', 'publicKeyIdentifier']
-    ensureFileDirectoryExists(config['cpiDatabaseConfig']['indexUrl'])
-    with open(config['cpiDatabaseConfig']['indexUrl'], 'w+') as output_file:
+    ensureFileDirectoryExists(config['cpiDatabaseConfig']['indexPath'])
+    with open(config['cpiDatabaseConfig']['indexPath'], 'w+') as output_file:
```
Comment (kate-harrison) [inline src/harness/testcases/WINNF_FT_S_WDB_testcase.py]: You need to change line 301 as well, then this looks great.

File: src/harness/testcases/WINNF_FT_S_WDB_testcase.py [hunk 6]
```diff
@@ -298,8 +298,8 @@ def test_WINNF_FT_S_WDB_2(self, config_filename):
     files = {('/' + cpi['publicKeyIdentifierFile']):
              cpi['publicKeyIdentifierFile']
              for cpi in config['cpiDatabaseConfig']['cpis']}
-    files['/' + config['cpiDatabaseConfig']['indexUrl']] = config[
-        'cpiDatabaseConfig']['indexUrl']
+    files['/' + config['cpiDatabaseConfig']['indexPath']] = config[
```
Comment (kate-harrison) [inline src/harness/testcases/WINNF_FT_S_WDB_testcase.py]: The first should be 'indexUrl' and the second should be 'indexPath', as described in database.py.

Comment (jni2000) [reply src/harness/testcases/WINNF_FT_S_WDB_testcase.py]: yes, should be good now. please take a look.

File: src/harness/testcases/WINNF_FT_S_WDB_testcase.py [hunk 7]
```diff
@@ -298,7 +298,7 @@ def test_WINNF_FT_S_WDB_2(self, config_filename):
     files = {('/' + cpi['publicKeyIdentifierFile']):
              cpi['publicKeyIdentifierFile']
              for cpi in config['cpiDatabaseConfig']['cpis']}
-    files['/' + config['cpiDatabaseConfig']['indexPath']] = config[
+    files['/' + config['cpiDatabaseConfig']['indexUrl']] = config[
```
Comment (kate-harrison) [inline src/harness/testcases/WINNF_FT_S_WDB_testcase.py]: Sorry I didn't catch it earlier, but the leading "/" should be removed from this line since it is already present in the index URL.

Comment (jni2000) [reply src/harness/testcases/WINNF_FT_S_WDB_testcase.py]: yes, you are right. thanks.
